src/renderer/components/+preferences/preferences.tsx

```diff
@@ -48,26 +46,40 @@ import { Kubernetes } from "./kubernetes";
 import { Editor } from "./editor";
 import { LensProxy } from "./proxy";
 import { Telemetry } from "./telemetry";
-import { Extensions } from "./extensions";
 import { sentryDsn } from "../../../common/vars";
+import { withInjectables } from "@ogre-tools/injectable-react";
+import type { InstalledExtension } from "../../../extensions/extension-discovery";
+import userExtensionsInjectable from "../+extensions/user-extensions/user-extensions.injectable";
+import { ExtensionSettingsPage } from "./extension-settings-page";
+import { Icon } from "../icon";
+import { uniqBy } from "lodash";
+
+interface Dependencies {
+  userExtensions: IComputedValue<InstalledExtension[]>;
+}
 
 @observer
-export class Preferences extends React.Component {
+class Preferences extends React.Component<Dependencies> {
   @observable historyLength: number | undefined;
 
-  constructor(props: {}) {
+  constructor(props: Dependencies) {
     super(props);
     makeObservable(this);
   }
 
   renderNavigation() {
-    const extensions = AppPreferenceRegistry.getInstance().getItems();
-    const telemetryExtensions = extensions.filter(e => e.showInPreferencesTab == "telemetry");
+    const preferenceRegistries = uniqBy(AppPreferenceRegistry.getInstance().getItems(), "extensionId");
```

Comment:
I think you should coordinate with @jim-docker, because he said that he was working to make app preferences injectable but this would be a perfect PR for them to be done in.

Reply:
I also think that this logic is a bit confusing and could be simplified.

Finally, why don't we just implement that extensions can add to any page in this PR too, it would be quite easy.